Editor/EditorCore/EditorGUIUtility.cs

```diff
@@ -311,6 +311,81 @@ public static float FreeSlider(GUIContent content, float value, float min, float
             return value;
         }
 
+        public static int FreeSliderWithRange(string content, int value, int min, int max, ref int uiMin, ref int uiMax, ref bool expanded)
+        {
+            slider_guicontent.text = content;
+            return FreeSliderWithRange(slider_guicontent, value, min, max, ref uiMin, ref uiMax, ref expanded);
+        }
+
+        /**
+         * Similar to EditorGUILayoutUtility.Slider, except this allows for values outside of the uiMin/uiMax bounds via the int field.
+         * Contrary to the FreeSlider however it has a hard range defined with min and max.
+         */
+        public static int FreeSliderWithRange(GUIContent content, int value, int min, int max, ref int uiMin, ref int uiMax, ref bool expanded)
+        {
+            float pixelsPerPoint = 1f;
+
+#if !UNITY_4_7 && !UNITY_5_0 && !UNITY_5_3
+            pixelsPerPoint = UnityEditor.EditorGUIUtility.pixelsPerPoint;
+#endif
+
+            float PAD = 8f / pixelsPerPoint;
+            const float SLIDER_HEIGHT = 16f;
+            const float MIN_LABEL_WIDTH = 0f;
+            const float MAX_LABEL_WIDTH = 128f;
+            const float MIN_FIELD_WIDTH = 48f;
+
+            GUILayoutUtility.GetRect(UnityEditor.EditorGUIUtility.currentViewWidth / pixelsPerPoint, 18);
+
+            Rect previousRect = GUILayoutUtility.GetLastRect();
+            float y = previousRect.y;
+
+            float labelWidth = content != null ? Mathf.Max(MIN_LABEL_WIDTH, Mathf.Min(GUI.skin.label.CalcSize(content).x + PAD, MAX_LABEL_WIDTH)) : 0f;
+            float remaining = ((Screen.width / pixelsPerPoint) - (PAD * 2f)) - labelWidth;
+            float sliderWidth = remaining - (MIN_FIELD_WIDTH + PAD);
+            float intWidth = MIN_FIELD_WIDTH;
+            float indentOffset = EditorGUI.indentLevel * 15f;
+
+            Rect labelRect = new Rect(PAD, y + 2f, labelWidth, SLIDER_HEIGHT);
+            Rect sliderRect = new Rect(labelRect.x + labelWidth, y + 1f, sliderWidth, SLIDER_HEIGHT);
+            Rect intRect = new Rect(sliderRect.x + sliderRect.width + PAD, y + 1f, intWidth, SLIDER_HEIGHT);
+            
+            Rect totalRect = GUILayoutUtility.GetRect(1, UnityEditor.EditorGUIUtility.singleLineHeight);            
+            Rect foldoutRect = new Rect(labelRect.xMax - 10, labelRect.y, 8, totalRect.height);
+
+            if (content != null)
+                GUI.Label(labelRect, content);
+
+            EditorGUI.BeginChangeCheck();
+
+            int controlID = GUIUtility.GetControlID(FocusType.Passive, sliderRect);
+            float tmp = value;
+            float tmpUIMin = uiMin;
+            float tmpUIMax = uiMax;
+            tmp = GUI.Slider(sliderRect, tmp, 0f, tmpUIMin, tmpUIMax, GUI.skin.horizontalSlider, (!EditorGUI.showMixedValue) ? GUI.skin.horizontalSliderThumb : "SliderMixed", true, controlID);
```

Comment:
is this the same as https://docs.unity3d.com/ScriptReference/GUI.HorizontalSlider.html?